libs/traits/src/lib.rs

```diff
@@ -281,56 +285,111 @@ pub trait CurrencyPrice<CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum CompoundingSchedule {
+	/// Interest compounds every second
+	Secondly,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum InterestRate<Rate> {
+	/// Interest accrues at a fixed rate
+	Fixed {
+		rate_per_year: Rate,
+		compounding: CompoundingSchedule,
+	},
+}
+
+impl<Rate> InterestRate<Rate> {
+	pub fn rate_per_year(&self) -> &Rate {
+		match self {
+			InterestRate::Fixed { rate_per_year, .. } => rate_per_year,
+		}
+	}
+}
+
+impl<Rate: EnsureAdd + EnsureSub> InterestRate<Rate> {
```

Comment:
Any reason why we are not implementeing the trait itself on the Type?

Reply:
You "cannot", this trait is implemented automatically for each T that already is `CheckedAdd`. You could implement `CheckedAdd` for this type, having automatically an `EnsureAdd` impl.

Reply:
Ah, right. Okay. Well not mandatory, but nice to have `CheckedAdd` then ^^

Reply:
Although I would like to do it. It's is required to be `PartialOrd` + `CheckedAdd` to be `EnsureAdd`, and it's difficult to determine a `PartialOrd` with arithmetic meaning for a type with several fields.

This is something that now, after a lot of usage, I would design differently. Instead of making this trait automatic for any `T`, I would implement it for basic & substrate types, allowing then any custom implementation for more complex custom time.

Maybe I do a new PR to Substrate this change. I think it can be beneficial, although it will be a huge breaking change, and I do not know if they would approve it.